src/commands/init.js

```diff
@@ -14,206 +14,241 @@ const { track } = require('../utils/telemetry')
+  }
+}
+
+const logExistingRepoSetupAndExit = ({ context, siteName, repoUrl }) => {
+  context.log()
+  context.log(chalk.underline.bold(`Success`))
+  context.log(`This site "${siteName}" is configured to automatically deploy via ${repoUrl}`)
+  // TODO add support for changing github repo in site:config command
+  context.exit()
+}
 
-    if (flags.manual) {
-      await configManual(this, siteInfo, repo)
+const configureGitHub = async ({ context, siteInfo, repo }) => {
+  try {
+    await configGithub(context, siteInfo, repo)
+  } catch (error) {
+    context.warn(`GitHub error: ${error.status}`)
+    if (error.status === 404) {
+      context.error(
+        `Does the repository ${repo.repo_path} exist and do you have the correct permissions to set up deploy keys?`,
+      )
     } else {
-      switch (repo.provider) {
-        case 'github': {
-          try {
-            await configGithub(this, siteInfo, repo)
-          } catch (error) {
-            this.warn(`GitHub error: ${error.status}`)
-            if (error.status === 404) {
-              this.error(
-                `Does the repository ${repo.repo_path} exist and do you have the correct permissions to set up deploy keys?`,
-              )
-            } else {
-              throw error
-            }
-          }
-          break
-        }
-        case 'gitlab':
-        default: {
-          this.error('No configurator found for the git hosting service')
-        }
-      }
+      throw error
     }
+  }
+}
 
-    // Success Message
-    this.log()
-    this.log(chalk.greenBright.bold.underline(`Success! Netlify CI/CD Configured!`))
-    this.log()
-    this.log(`This site is now configured to automatically deploy from ${repo.provider} branches & pull requests`)
-    this.log()
-    this.log(`Next steps:
+const configureProvider = async ({ context, siteInfo, repo }) => {
+  switch (repo.provider) {
+    case 'github': {
+      await configureGitHub({ context, siteInfo, repo })
+      break
+    }
+    default: {
+      context.error('No configurator found for the git hosting service')
+    }
+  }
+}
+
+const logSuccess = ({ context, repo }) => {
+  // Success Message
+  context.log()
+  context.log(chalk.greenBright.bold.underline(`Success! Netlify CI/CD Configured!`))
+  context.log()
+  context.log(`This site is now configured to automatically deploy from ${repo.provider} branches & pull requests`)
+  context.log()
+  context.log(`Next steps:
 
   ${chalk.cyanBright.bold('git push')}       Push to your git repository to trigger new site builds
   ${chalk.cyanBright.bold('netlify open')}   Open the Netlify admin URL of your site
   `)
+}
+
+class InitCommand extends Command {
+  async run() {
+    const { flags } = this.parse(InitCommand)
+    const { site, state } = this.netlify
+    let { siteInfo } = this.netlify
+
+    // Check logged in status
+    await this.authenticate()
+
+    await reportAnalytics({ context: this, flags })
```

Comment:
Should we directly pass the `context` property that each function is actually using instead of the full object? This would make it clearer what the function actually needs, and might remove the need for a context object altogether (or minimize it).

For example, `reportAnalytics()` only needs the `config`. And some others only need the `log` helper function.

Reply:
Good point. Let me fix that.

Reply:
Fixed in https://github.com/netlify/cli/pull/1614/commits/2fd92a455bc81f22f5609a2a46357aad1d27585e